lms/djangoapps/program_enrollments/api/v1/serializers.py

```diff
@@ -40,3 +40,26 @@ class Meta(object):
 
     def get_account_exists(self, obj):
         return bool(obj.user)
+
+
+class InvalidStatusMixin(object):
+    """
+    Mixin to provide has_invalid_status method
+    """
+    def has_invalid_status(self):
+        """
+        Returns whether or not this serializer has an invlaid error choice on the "status" field
+        """
+        invalid_status = 'status' in self.errors
+        return invalid_status and self.errors['status'][0].code == 'invalid_choice'
```

Comment:
I am confused why look at the first status within the self.errors? Also, this set of code is confusing to read. I would change it to
```
error_statuses = self.errors.get('status')
return error_statuses and getattr(error_statuses[0], 'code') == 'invalid_choice'
```